subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublication.java

```diff
@@ -125,14 +119,8 @@ public DefaultIvyPublication(
         this.versionMappingStrategy = versionMappingStrategy;
         this.taskDependencyFactory = taskDependencyFactory;
 
-        IvyComponentParser ivyComponentParser = new IvyComponentParser(
-            instantiator,
-            platformSupport,
-            projectDependencyResolver,
-            ivyArtifactNotationParser,
-            documentationRegistry,
-            versionMappingStrategy,
-            collectionCallbackActionDecorator
+        IvyComponentParser ivyComponentParser = objectFactory.newInstance(
```

Comment:
The version mapping strategy is a per-publication type. I could see an approach where we don't pass the version mapping strategy to this type upon construction, but instead only pass it to the methods that need it, but for now I think its fine to keep this as-is.

Applies to other similar comments

Reply:
I was able to convert the `DependencyCoordinateResolverFactory` to a service. The `IvyComponentParser` and `MavenComponentParser` are a little more difficult, since they rely on the artifact parsers which are also currently not services. 

I think it is a good idea to do this conversion, but it required a bit more changes than I wanted to do in this PR. 

I think its worth cleaning up a lot of stuff in this area related to this -- like the property comment you mentioned here